pkg/apis/kubernetes/v1/cluster_genclient.go

```diff
@@ -2,59 +2,10 @@ package v1
 
 import (
 	"context"
-	"errors"
 	"net/url"
-
-	"go.anx.io/go-anxcloud/pkg/api/types"
-	"go.anx.io/go-anxcloud/pkg/utils/pointer"
 )
 
-// ErrManagedPrefixSet is returned if a prefix is both set to managed and also set
-// to an existing prefix on create
-var ErrManagedPrefixSet = errors.New("managed prefixes cannot be set on create")
-
 // EndpointURL returns the common URL for operations on Cluster resource
 func (c *Cluster) EndpointURL(ctx context.Context) (*url.URL, error) {
 	return endpointURL(ctx, c, "/api/kubernetes/v1/cluster.json")
 }
-
-// explicitlyFalse returns true if the value of the provided
-// bool pointer is set to false, nil and true pointer return false
-func explicitlyFalse(b *bool) bool {
-	return b != nil && !pointer.BoolVal(b)
-}
-
-// prefixConfigurationValidCreate validates that prefixes configured as managed
-// are not set. This is only relevant for `Create` operations.
-func prefixConfigurationValidCreate(c *Cluster) bool {
-	var (
-		internalV4 = c.InternalIPv4Prefix == nil || explicitlyFalse(c.ManageInternalIPv4Prefix)
-		externalV4 = c.ExternalIPv4Prefix == nil || explicitlyFalse(c.ManageExternalIPv4Prefix)
-		externalV6 = c.ExternalIPv6Prefix == nil || explicitlyFalse(c.ManageExternalIPv6Prefix)
-	)
-
-	return internalV4 && externalV4 && externalV6
-}
```

Comment:
TODO: should probably go into a different hook (maybe a new RequestValidationHook)